core/src/main/java/org/elasticsearch/index/query/TermsQueryBuilder.java

```diff
@@ -215,4 +434,109 @@ public void doXContent(XContentBuilder builder, Params params) throws IOExceptio
     public String getWriteableName() {
         return NAME;
     }
+
+    @Override
+    protected Query doToQuery(QueryShardContext context) throws IOException {
+        List<Object> terms;
+        if (isTermsLookupQuery()) {
+            terms = context.handleTermsLookup(termsLookup.toTermsLookup(context));
+        } else {
+            terms = new ArrayList<>();
+            terms.addAll(values);
+        }
+        if (terms.isEmpty()) {
+            return Queries.newMatchNoDocsQuery();
+        }
+        return handleTermsQuery(terms, fieldName, context, minimumShouldMatch, disableCoord);
+    }
+
+    static Query handleTermsQuery(List<Object> terms, String fieldName, QueryShardContext context, String minimumShouldMatch, boolean disableCoord) {
```

Comment:
Is this a separate method for future use elsewhere? I only see it beeing used in doToQuery(), so in that case why not merge it?